packages/gatsby/cache-dir/app.js

```diff
@@ -13,6 +13,12 @@ import pages from "./pages.json"
 window.___emitter = emitter
 setApiRunnerForLoader(apiRunner)
 
+// necessary for hot-reloading of react hooks
+setConfig({
+  ignoreSFC: true,
```

Comment:
This option, without patches in React would disable RHL for any SFC at all.
Ie anything below any SFC would lose state. Probably this is not the goal.

Use `pureSFC` instead.

Reply:
```suggestion
  pureSFC: true,
```

you mean?

Reply:
👍 👍 👍 👍 
but look like `pureSFC` option is a bit broken in 4.5.1 - https://github.com/gaearon/react-hot-loader/issues/1115